astropy/coordinates/sky_coordinate.py

```diff
@@ -1359,11 +1360,29 @@ def radial_velocity_correction(self, kind='barycentric', obstime=None,
                              "'{}'".format(kind))
 
         gcrs_p, gcrs_v = location.get_gcrs_posvel(obstime)
-        gtarg = self.transform_to(GCRS(obstime=obstime,
-                                       obsgeoloc=gcrs_p,
-                                       obsgeovel=gcrs_v))
-        targcart = gtarg.represent_as(UnitSphericalRepresentation).to_cartesian()
-        return targcart.dot(v_origin_to_earth + gcrs_v)
+        # transforming to GCRS is not the correct thing to do here, since we don't want to
+        # include aberration (or light deflection)? Instead, only apply parallax if necessary
+        if self.data.__class__ is UnitSphericalRepresentation:
+            targcart = self.represent_as(UnitSphericalRepresentation).to_cartesian()
+        else:
+            # skycoord has distances so apply parallax
+            obs_icrs_cart = get_body_barycentric('earth', obstime) + gcrs_p
+            icrs_cart = self.cartesian
```

Comment:
Don't we have to transform ourselves to ICRS for this to hold?

Reply:
damn - we do! This was a bug in the original version too. Also needs conversion in the ```if``` part of the ```if...else...``` clause.